src/PointwiseFunctions/AnalyticSolutions/Poisson/Moustache.cpp

```diff
@@ -0,0 +1,76 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/Poisson/Moustache.hpp"
+
+#include <cmath>
+#include <pup.h>  // IWYU pragma: keep
+
+#include "DataStructures/DataVector.hpp"     // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+// IWYU pragma: no_forward_declare Tensor
+
+namespace Poisson {
+namespace Solutions {
+
+template <size_t Dim>
+tuples::TaggedTuple<Field> Moustache<Dim>::field_variables(
+    const tnsr::I<DataVector, Dim>& x) const noexcept {
+  auto field = make_with_value<Scalar<DataVector>>(x, 1.);
+  for (size_t d = 0; d < Dim; d++) {
+    field.get() *= x.get(d) * (1. - x.get(d));
+  }
+  auto norm_square = make_with_value<DataVector>(get<0>(x), 0.);
+  for (size_t d = 0; d < Dim; d++) {
+    norm_square += square(x.get(d) - 0.5);
+  }
+  field.get() *= pow(norm_square, 3. / 2.);
+  return {std::move(field)};
+}
+
+template <>
+tuples::TaggedTuple<::Tags::Source<Field>> Moustache<1>::source_variables(
+    const tnsr::I<DataVector, 1>& x) const noexcept {
+  auto source = make_with_value<Scalar<DataVector>>(x, 1.);
+  source.get() *= (0.875 - 8.5 * get<0>(x) + 28.5 * square(get<0>(x)) -
+                   40. * pow(get<0>(x), 3) + 20. * pow(get<0>(x), 4)) /
+                  abs(get<0>(x) - 0.5);
+  return {std::move(source)};
+}
+
+template <>
+tuples::TaggedTuple<::Tags::Source<Field>> Moustache<2>::source_variables(
+    const tnsr::I<DataVector, 2>& x) const noexcept {
+  auto source = make_with_value<Scalar<DataVector>>(x, 1.);
+  auto x1 = get<0>(x);
+  auto x2 = get<1>(x);
+  source.get() *= (6. * pow(x1, 5) - 2. * pow(x1, 6) +
+                   pow(x1, 4) * (-11. + 39. * x2 - 39. * pow(x2, 2)) +
+                   pow(x1, 3) * (12. - 78. * x2 + 78. * pow(x2, 2)) +
+                   pow(x1, 2) * (-7. + 65.5 * x2 - 104.5 * pow(x2, 2) +
+                                 78. * pow(x2, 3) - 39. * pow(x2, 4)) +
+                   x1 * (2. - 26.5 * x2 + 65.5 * pow(x2, 2) - 78. * pow(x2, 3) +
+                         39. * pow(x2, 4)) +
+                   x2 * (2. - 7. * x2 + 12. * pow(x2, 2) - 11. * pow(x2, 3) +
+                         6. * pow(x2, 4) - 2. * pow(x2, 5)));
+  auto norm_square = make_with_value<DataVector>(get<0>(x), 0.);
+  for (size_t d = 0; d < 2; d++) {
+    norm_square += square(x.get(d) - 0.5);
+  }
+  source.get() /= sqrt(norm_square);
+  return {std::move(source)};
+}
+
+template <size_t Dim>
+void Moustache<Dim>::pup(PUP::er& /*p*/) noexcept {}
+
+}  // namespace Solutions
+}  // namespace Poisson
+
+template class Poisson::Solutions::Moustache<1>;
+template class Poisson::Solutions::Moustache<2>;
```

Comment:
Should document somewhere that there is no Moustache<3>

Reply:
The documentation now describes the 1D and 2D solution